### PR TITLE
feature: gpi gate

### PR DESCRIPTION
**Pull request by ajberdy**
*Issue #, if available:*

*Description of changes:*

Add GPi gate with numerical differentiation method

*Testing done:*

Unit testing and manual

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-pennylane-plugin-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ajberdy** [comment]
Idk why the docs build failed, works locally and error message is unhelpful.

File: src/braket/pennylane_plugin/ops.py [hunk 1]
```diff
@@ -308,3 +306,47 @@ def compute_matrix(phi):
...
+        return np.array(
+            [
+                [0, np.exp(-1j * phi)],
+                [np.exp(1j * phi), 0],
+            ]
+        )
+
+    def adjoint(self):
+        (phi,) = self.parameters
+        return GPi(phi, wires=self.wires)
```
Comment (speller26) [inline src/braket/pennylane_plugin/ops.py:352]: Don't we need to add `np.pi` here?

Comment (ajberdy) [reply src/braket/pennylane_plugin/ops.py:352]: Nope, that's GPi2

Comment (speller26) [reply src/braket/pennylane_plugin/ops.py:352]: Oh whoops, forgot about that

File: src/braket/pennylane_plugin/translation.py [hunk 2]
```diff
@@ -318,6 +321,12 @@ def _(zz: qml.IsingZZ, parameters):
     return gates.ZZ(-phi) if zz.inverse else gates.ZZ(phi)
 
 
+@_translate_operation.register
+def _(gpi: GPi, parameters):
+    phi = parameters[0]
+    return gates.GPi(phi + np.pi) if gpi.inverse else gates.GPi(phi)
```
Comment (speller26) [inline src/braket/pennylane_plugin/translation.py:327]: Why the np.pi here then?

Comment (ajberdy) [reply src/braket/pennylane_plugin/translation.py:327]: 🐛

**Review by speller26 — APPROVED** [review]

**Pull request merged**